tests/drum/test_fit.py

```diff
@@ -827,11 +827,13 @@ def test_fit_schema_failure(
             assert_if_fail=False,
         )
 
-        # The predict server will not return the full stacktrace since it is ran in a forked process
         if error_in_predict_server:
             assert (
                 "Schema validation found mismatch between output dataset and the supplied schema"
                 in stdout
             )
         else:
-            assert "DrumSchemaValidationException" in stderr
+            assert (
+                "Schema validation found mismatch between input dataset and the supplied schema"
+                in stderr
+            )
```

Comment:
Explicitly exits (e.g., equivalent with raising SystemExit) seems to change the stdout/stderr contents. The following stacktrace (case of `(PYTORCH, BINARY, PYTHON, False)`) in `stderr` was changed:

- Before
```
'2021-11-03 13:04:27,375 ERROR drum.datarobot_drum.drum.typeschema_validation:  Schema validation found mismatch between input dataset and the supplied schema\n2021-11-03 13:04:27,375 ERROR drum.datarobot_drum.drum.typeschema_validation:  Datatypes incorrect. Data has types: CAT, NUM, TXT, which includes values that are not in NUM.\n2021-11-03 13:04:27,375 WARNING drum:  \x1b[33mLooks like there is a problem. To get more output information try to run with: --verbose\x1b[0m\nTraceback (most recent call last):\n  File "/home/lyndon/.virtualenvs/datarobot-drum/bin/drum", line 7, in <module>\n    exec(compile(f.read(), __file__, \'exec\'))\n  File "/home/lyndon/workspace/datarobot-user-models/custom_model_runner/bin/drum", line 6, in <module>\n    main()\n  File "/home/lyndon/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/main.py", line 107, in main\n    CMRunner(runtime).run()\n  File "/home/lyndon/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/drum.py", line 469, in run\n    self.run_fit()\n  File "/home/lyndon/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/drum.py", line 506, in run_fit\n    self.schema_validator.validate_inputs(input_data)\n  File "/home/lyndon/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/typeschema_validation.py", line 540, in validate_inputs\n    return self._run_validate(dataframe, self._input_validators, "input")\n  File "/home/lyndon/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/typeschema_validation.py", line 568, in _run_validate\n    "schema validation failed for {}:\\n {}".format(step_label, errors)\ndatarobot_drum.drum.exceptions.DrumSchemaValidationException: schema validation failed for input:\n [\'Datatypes incorrect. Data has types: CAT, NUM, TXT, which includes values that are not in NUM.\']\n'
```

- After 
```
'2021-11-03 14:37:22,238 ERROR drum.datarobot_drum.drum.typeschema_validation:  Schema validation found mismatch between input dataset and the supplied schema\n2021-11-03 14:37:22,238 ERROR drum.datarobot_drum.drum.typeschema_validation:  Datatypes incorrect. Data has types: CAT, NUM, TXT, which includes values that are not in NUM.\n2021-11-03 14:37:22,238 WARNING drum:  \x1b[33mLooks like there is a problem. To get more output information try to run with: --verbose\x1b[0m\n'
```